compiler-rt/lib/scudo/standalone/combined.h

```diff
@@ -1564,11 +1570,11 @@ class Allocator {
                   UINTPTR_MAX);
```

Comment:
This doesn't belong to StackDepot, this belongs here to make sure that even with the biggest possible `TabSize` and `RingSize` we don't overflow our 64 bit integer here.

Reply:
There's nothing that relates to any constexpr arguments or variables here. That's the constraint to the `StackDepot` (especially the ` static_assert(sizeof(StackDepot) % alignof(atomic_u64) == 0)`)

Besides, I'm not sure if the overflow is intended,
```
UINT32_MAX * sizeof(atomic_u64) * UINT32_MAX * sizeof(atomic_u32)
```
I guess the assertion will never be triggered.

Reply:
If you read the comment, this assumes the max RingSize and TabSize, and the static_assert proves that we don't need to bother with overflow logic because it can never overflow

Reply:
IncrementalCompilerBuilder

Reply:
(except of course it doesn't work, because that will also overflow. let me remove it for now).

Reply:
I mean, from the statements,
```
static_assert(sizeof(StackDepot) % alignof(atomic_u64) == 0);
```
```
static_assert(sizeof(StackDepot) + UINT32_MAX * sizeof(atomic_u64) *
                                           UINT32_MAX * sizeof(atomic_u32) < UINTPTR_MAX);
```
They are checking things independent to `RingSize` and `TabSize` here. The checks don't need the value of two variables. Any values of RingSize/TabSize will static assert the same thing. They are general constraints for StackDepot.

The expression results in a number larger than uint32_t but it'll wrap-around and the value will always be in the range of uint32_t. When it compares with UINTPTR_MAX, it's almost always true on 32 bit and always true on 64 bit. static_assert doesn't help with this.

In addition, even the `StackDepot meets the requirement, you still can't allocate ring buffer with UINT32_MAX. This seems to be an unreasonable assertion.

Reply:
Just did a quick glance on this, another problem in that expression is that we may want to change it from
```
sizeof(StackDepot) + UINT32_MAX * sizeof(atomic_u64) * UINT32_MAX * sizeof(atomic_u32)
```
to
```
sizeof(StackDepot) + UINT32_MAX * sizeof(atomic_u64)  + UINT32_MAX * sizeof(atomic_u32)
```
which change the middle "*" to "+". But still, the UINT32_MAX may not be a good boundary.